duckdb-vortex/src/vortex_scan.cpp

```diff
@@ -273,6 +272,7 @@ static void CreateScanPartitions(ClientContext &context, const BindData &bind, S
 		};
 
 		if (PinFileToThread(global_state)) {
```

Comment:
You can pull out this check (before the loop) to avoid reevaluating.

Also assign the value once out there?

Reply:
Well we did have the out if once. But sure we can add an additional one.